### ISSUE TITLE
Error with `broadcast_in_dim` and `slice` resulting in a segment not being compiled without prior error!

### ISSUE THREAD

**Issue by kevinstephano** [NVIDIA/Fuser#851]
The error from this bug appears like a segment is not compiled to the end user which is really bad!

# Repro

```python
import torch

from nvfuser import FusionDefinition, DataType

def nvfuser_fusion_id2889(fd : FusionDefinition) -> None :
    T0 = fd.define_tensor(shape=[1, -1, -1], contiguity=[None, True, True], dtype=DataType.Half, is_cpu=False)
    S3 = fd.define_scalar(1, dtype=DataType.Int)
    S4 = fd.define_scalar(1, dtype=DataType.Int)
    S5 = fd.define_scalar(1, dtype=DataType.Int)
    S6 = fd.define_scalar(1, dtype=DataType.Int)
    S7 = fd.define_scalar(5, dtype=DataType.Int)
    S8 = fd.define_scalar(3, dtype=DataType.Int)
    S9 = fd.define_scalar(1, dtype=DataType.Int)
    S10 = fd.define_scalar(1, dtype=DataType.Int)
    V11 = fd.define_vector([S3, S4, S5, S6, S7, S8, S9, S10], dtype=DataType.Int)
    T12 = fd.ops.broadcast_in_dim(T0, shape=V11, broadcast_dims=[2, 4, 5])
    T13 = fd.ops.slice(T12, start_indices=[0, 0, 0, 0, 2, 0, 0, 0], end_indices=[1, 1, 1, 1, 3, 3, 1, 1], strides=[1, 1, 1, 1, 1, 1, 1, 1])
    #T14 = fd.ops.squeeze(T13, original_shape=[1, 1, 1, 1, 1, 3, 1, 1], dims=[2, 4])
    fd.add_output(T13)

with FusionDefinition() as fd:
    nvfuser_fusion_id2889(fd)

inputs = [
    torch.randn((15,), dtype=torch.float16, device='cuda:0').as_strided((1, 5, 3), (15, 3, 1)),
]
fd.execute(inputs)
```

# Error with `segmented_fusion` debug information enabled (I think this is the real error)
```
Segmented_Fusion{
groups:
g{0, 3}

g{7}

edges:
e{ g{0, 3}
 -> g{7}
(T2_g[ bS11{1}, bS12{1}, bS13{1}, bS14{1}, iS15{( (nvfuser_index_t)(5) )}, iS16{( (nvfuser_index_t)(3) )}, bS17{1}, bS18{1} ]) }


group details:
g{(pointwise)
inputs:
T0_g[ bS0{1}, iS1{i1}, iS2{i2} ] __half
outputs:
T2_g[ bS11{1}, bS12{1}, bS13{1}, bS14{1}, iS15{( (nvfuser_index_t)(5) )}, iS16{( (nvfuser_index_t)(3) )}, bS17{1}, bS18{1} ] __half


T1_l[ bS3{1}, bS4{1}, bS5{1}, bS6{1}, iS7{i1}, iS8{i2}, bS9{1}, bS10{1} ]
   = broadcast( T0_g[ bS0{1}, iS1{i1}, iS2{i2} ] )
(0)
T2_g[ bS11{1}, bS12{1}, bS13{1}, bS14{1}, iS15{( (nvfuser_index_t)(5) )}, iS16{( (nvfuser_index_t)(3) )}, bS17{1}, bS18{1} ]
   = Set( T1_l[ bS3{1}, bS4{1}, bS5{1}, bS6{1}, iS7{i1}, iS8{i2}, bS9{1}, bS10{1} ] )
(3)
}

g{(pointwise)
inputs:
T2_g[ bS11{1}, bS12{1}, bS13{1}, bS14{1}, iS15{( (nvfuser_index_t)(5) )}, iS16{( (nvfuser_index_t)(3) )}, bS17{1}, bS18{1} ] __half
outputs:
T3_g[ bS19{1}, bS20{1}, bS21{1}, bS22{1}, bS24{1}rf, iS25{( (nvfuser_index_t)(3) )}rf, bS26{1}, bS27{1} ] __half


T3_g[ bS19{1}, bS20{1}, bS21{1}, bS22{1}, bS24{1}rf, iS25{( (nvfuser_index_t)(3) )}rf, bS26{1}, bS27{1} ]
   = slice( T2_g[ bS11{1}, bS12{1}, bS13{1}, bS14{1}, iS15{( (nvfuser_index_t)(5) )}, iS16{( (nvfuser_index_t)(3) )}, bS17{1}, bS18{1} ], { {0, 1, 1} {0, 1, 1} {0, 1, 1} {0, 1, 1} {2, 3, 1} {0, 3, 1} {0, 1, 1} {0, 1, 1} } )
(7)
}

} //Segmented_Fusion

[E thread_pool.cpp:110] Exception in thread pool task: it != broadcast_origin_map_.end() INTERNAL ASSERT FAILED at "/opt/pytorch/nvfuser/csrc/device_lower/analysis/trivial_broadcast.cpp":98, please report a bug with repro script to NVFuser at https://github.com/NVIDIA/Fuser/issues. Broadcast origin info not found for
 producer broadcast domain: bS24{1}rf of T5_l[ bS19{1}, bS20{1}, bS21{1}, bS22{1}, bS24{1}rf, iS25{( (nvfuser_index_t)(3) )}rf, bS26{1}, bS27{1} ]
Exception raised from dispatch at /opt/pytorch/nvfuser/csrc/device_lower/analysis/trivial_broadcast.cpp:98 (most recent call first):
```

# Error reported
```
Traceback (most recent call last):
  File "/workspace/lightning-thunder/test_2.py", line 27, in <module>
    fd.execute(inputs)
  File "/opt/pytorch/nvfuser/nvfuser/__init__.py", line 122, in execute
    result = self._execute(
RuntimeError: executors_.at(group_id).isCompiled() INTERNAL ASSERT FAILED at "/opt/pytorch/nvfuser/csrc/kernel_cache.cpp":1192, please report a bug with repro script to NVFuser at https://github.com/NVIDIA/Fuser/issues.
Exception raised from getKernelConfig at /opt/pytorch/nvfuser/csrc/kernel_cache.cpp:1192 (most recent call first)
```

**Comment by naoyam** [comment]
The issue here is:

```
it != broadcast_origin_map_.end() INTERNAL ASSERT FAILED at "/opt/pytorch/nvfuser/csrc/device_lower/analysis/trivial_broadcast.cpp":98, please report a bug with repro script to NVFuser at https://github.com/NVIDIA/Fuser/issues. Broadcast origin info not found for
 producer broadcast domain: bS24{1}rf of T5_l[ bS19{1}, bS20{1}, bS21{1}, bS22{1}, bS24{1}rf, iS25{( (nvfuser_index_t)(3) )}rf, bS26{1}, bS27{1} ]
Exception raised from dispatch at /opt/pytorch/nvfuser/csrc/device_lower/analysis/trivial_broadcast.cpp:98 (most recent call first):
```

@jacobhinkle also observed a similar error. Do you have anything for this, @jacobhinkle?

**Comment by jacobhinkle** [comment]
I think this is a duplicate of #596. I checked that the repro above works when using the branch from #610.

**Comment by jacobhinkle** [comment]
BTW, regarding the error messages: this happens because of parallel compilation. If you run with `NVFUSER_DISABLE=parallel_compile`, you will get the "clean" error message.

**Comment by jacobhinkle** [comment]
I believe this was fixed by #610 as the repro runs fine for me now.
